src/store/channels/saga.ts

```diff
@@ -220,6 +221,15 @@ export function* receivedRoomMemberPowerLevelChanged(action) {
   yield call(receiveChannel, { id: roomId, moderatorIds });
 }
 
+export function* onSetReadReceiptPreference(action) {
+  const { preference } = action.payload;
+  try {
+    yield call(setReadReceiptPreference, preference);
+  } catch (error) {
```

Comment:
Suggestion: This should probably translate between a "Domain" enum and the Matrix enum (or a different option: client wrapper should maybe have "setToPrivate" and "setToPublic" functions). The reason is, you don't really want your UI components to have to import Matrix enumerations in order to do their work. Those should only have knowledge of "Domain" information.